### PR TITLE
CODEOWNERS: Overhaul in order to enable required reviews

### PR DESCRIPTION
**Pull request by carlescufi**
In preparation for the enabling of the GitHub "Require reviews from Code Owners" branch protection setting, reorganize the Codeowners file so that Vestavind is aware of changes in the files it needs to oversee.

See also: https://nordicsemi.atlassian.net/wiki/spaces/SPG/pages/635604878/CODEOWNERS+file

### CODE REVIEW

**Review by rlubos — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

File: CODEOWNERS
Comment (greg-fer) [inline CODEOWNERS:77]: These I'm watching because of my involvement with Weyland and the ongoing restructuring. Please keep me explicitly on them. Thanks.

File: CODEOWNERS
Comment (tmon-nordic) [reply CODEOWNERS:54]: It will request review from all tech writers (each tech writer will get email notification; at least in default notification settings), but review from only one tech writer satisfies the requirement and would allow the PR to be merged (if all other required teams have approved).

File: CODEOWNERS
Comment (carlescufi) [reply CODEOWNERS:77]: But you are there, because of the following line:
`/doc/**/*.rst @nrfconnect/ncs-doc-owners`

Comment (greg-fer) [reply CODEOWNERS:77]: Is there a way I can filter notifications and additions just for these four categories above? I work with 5 different teams in the NCS, and I don't want to be added to doc PRs of the other 34.

File: CODEOWNERS
Comment (tmon-nordic) [reply CODEOWNERS:54]: Should each NCS team have a subteam for doc where the tech-writer that works with the team would be added? Or does this sound like an overkill?

It looks that we would need to have the generic /doc/**/*.rst at top, and then the specific entries (either with individual tech writes or the subteams) below. If the specific entries catch all .rst files, then the generic won't hit and therefore only the individuals will be notified.

Is there any chance to organize the doc directory with NCS teams subdirectories? Or would such change result in horrible mess?

Comment (greg-fer) [reply CODEOWNERS:54]: > Should each NCS team have a subteam for doc where the tech-writer that works with the team would be added? Or does this sound like an overkill?

So far, we've always worked based on the assumption that each team knows the tech writers working with them and knows when to add them ([source 1](https://nordicsemi.atlassian.net/wiki/spaces/NCS/pages/108225083/Documentation+process#Documentationprocess-Sampleandapplicationdocumentation) , [source 2](https://nordicsemi.atlassian.net/wiki/spaces/TECHDOC/pages/120302385/NCS+team+assignment) ). So from this perspective, the proposed changes sound like an overkill.

You really need tech writers in the codeowners file when one needs to monitor changes across different areas. For example, one of the reasons I asked to be added because I'm working with changes to the doc structure, so I need a cross-team overview to general sections.

Maybe a better option than having 13 tech writers in doc-owners is to have some of them in the group (3? 4?), and this group would ensure that the relevant tech writers are added to the PRs individually. But I still feel like we're reinventing the wheel for a process that is working relatively well (teams adding tech writers based on the need).

> Is there any chance to organize the doc directory with NCS teams subdirectories? Or would such change result in horrible mess?

The doc directory in sdk-nrf? No. A tech writer on Matter could contribute to apps/matter_bridge, protocols/matter, samples/matter, which would result in a spaghetti-style horrible mess of duplicated directories.

Comment (carlescufi) [reply CODEOWNERS:56]: @greg-fer can we perhaps start with this, and then refine it as we go? Because we will discover a bunch of things the moment we merge this and enable required reviews by code owners.

File: CODEOWNERS
Comment (carlescufi) [reply CODEOWNERS:77]: I am really not sure about that. You probably need to use an email filter or the built-in GitHub web notifications mechanism for that.

Comment (tmon-nordic) [reply CODEOWNERS:77]: GitHub web notification cannot really filter. The only routing it can do is to direct messages to additional email addresses based on organization. Filtering in emails is probably not going to work, because the PR titles are rather hard target to get reliable filter on.

File: CODEOWNERS
Comment (greg-fer) [reply CODEOWNERS:56]: Apologies, didn't want to block this. I trust your judgment, @carlescufi . There was also @b-gent who had concerns though, and I know from some colleagues in TechPubs that they were also pulled into PRs out of the blue and had no idea why.

**Comment by lemrey** [comment]
@carlescufi I pushed a commit here to add the `ncs-modem` team where relevant.
@grochu, I am still code-owner of `esb`, is this something your team could take responsibility for?

File: CODEOWNERS
Comment (endre-nordic) [reply CODEOWNERS:153]: Not just this line. Search and remove from all paths

File: CODEOWNERS
Comment (plskeggs) [inline CODEOWNERS:82]: Please replace @plskeggs @jayteemo @glarsennordic with @nrfconnect/ncs-nrf-cloud.

File: CODEOWNERS
Comment (plskeggs) [inline CODEOWNERS:188]:
```suggestion
/samples/cellular/nrf_cloud_*             @nrfconnect/ncs-nrf-cloud
```

File: CODEOWNERS
Comment (plskeggs) [inline CODEOWNERS:82]:
```suggestion
/include/net/nrf_cloud_*                  @nrfconnect/ncs-nrf-cloud
```

File: CODEOWNERS
Comment (plskeggs) [inline CODEOWNERS:286]:
```suggestion
/subsys/net/lib/nrf_cloud/                @nrfconnect/ncs-nrf-cloud
```

Comment (plskeggs) [reply CODEOWNERS:286]: Please also change:
/tests/subsys/net/lib/nrf_cloud/ @tony-le-24
to:
/tests/subsys/net/lib/nrf_cloud/ @nrfconnect/ncs-nrf-cloud

**Review by plskeggs — CHANGES_REQUESTED** [review]
Change to @nrfconnect/ncs-nrf-cloud, thanks.

If you'd prefer that I do this as a separate PR, no problem.

**Pull request closed without merging**
